### PR TITLE
Payment methods

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

This PR represents a larger body of work to bring payment methods to settings.

# 🤔 Why

This allows users to add/edit/delete payment methods in the app.

# 🛠 How

https://github.com/kickstarter/ios-oss/pulls?utf8=%E2%9C%93&q=is%3Apr+payment-methods+

### CODE REVIEW

File: Kickstarter-iOS/DataSources/PaymentMethodsDataSource.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+             cellClass: CreditCardCell.self,
+             inSection: 0)
+  }
+
+  override func configureCell(tableCell cell: UITableViewCell, withValue value: Any) {
+    switch (cell, value) {
+    case let (cell as CreditCardCell, value as GraphUserCreditCard.CreditCard):
+      cell.configureWith(value: value)
+    default:
+      assertionFailure("Unrecognized (cell, viewModel) combo.")
```
Comment (dusi) [inline Kickstarter-iOS/DataSources/PaymentMethodsDataSource.swift]: We sure we want to crash if something goes wrong?

Comment (dusi) [reply Kickstarter-iOS/DataSources/PaymentMethodsDataSource.swift]: Can someone double check that we're not currently testing this behaviour?

Comment (justinswart) [reply Kickstarter-iOS/DataSources/PaymentMethodsDataSource.swift]: This is in all of our data sources, there isn't really an alternative 🤷‍♂️ . Will `assertionFailure` crash in production though? Thought asserts are ignored if not running in debug.

File: Kickstarter-iOS/Views/Cells/CreditCardCell.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import KsApi
+import Library
+import Prelude
+import ReactiveSwift
```
Comment (dusi) [inline Kickstarter-iOS/Views/Cells/CreditCardCell.swift]: Looks like this import is not necessary. Maybe we can remove it in the future (should not block the release)

File: Kickstarter-iOS/Views/Controllers/AddNewCardViewController.swift [hunk 3]
```diff
@@ -0,0 +1,299 @@
+import KsApi
+import Library
+import Prelude
+import ReactiveSwift
```
Comment (dusi) [inline Kickstarter-iOS/Views/Controllers/AddNewCardViewController.swift]: Same as above

File: KsApi/models/GraphUserCreditCardTests.swift [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                  "lastFour": "1111",
+                  "id": "2768",
+                }
+            ],
+            "totalCount": 0
+          }
+        }
+        """
+
+    let data = jsonString.data(using: .utf8)
```
Comment (justinswart) [inline KsApi/models/GraphUserCreditCardTests.swift]: In future we should try to replace these with the non-failable initializer:
```swift
let data = Data(bytes: jsonString.utf8)
```
Then we won't need the force unwraps.

File: Library/ViewModels/AddNewCardViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  }
+
+  private let stripeErrorProperty = MutableProperty(Error?.none)
+  public func stripeError(_ error: Error?) {
+    self.stripeErrorProperty.value = error
+  }
+
+  private let stripeTokenProperty = MutableProperty<(String, String)?>(nil)
+  public func stripeCreated(_ token: String?, stripeID: String?) {
+    if let token = token, let stripeID = stripeID {
```
Comment (justinswart) [inline Library/ViewModels/AddNewCardViewModel.swift]: Our inputs should not contain any logic.

File: KsApi/models/DeletePaymentMethodEnvelopeTests.swift [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                ]
+              }
+            }
+        }
+      }
+      """
+    let data = jsonString.data(using: .utf8)
+
+    do {
+      //swiftlint:disable force_unwrapping
```
Comment (dusi) [inline KsApi/models/DeletePaymentMethodEnvelopeTests.swift]: With a lot of these disables we do not re-enable them after the interested scope. This applies to some of the other rules in this PR. Should we use `<disable><enable>` or `<disable-next-line>`?

**Pull request merged**